contracts/governance/token/WrappedGovernanceToken.sol

```diff
@@ -0,0 +1,555 @@
+     * @param expiry Ordered array of timestamp at which the signatures expire.
+     * @param v Ordered array of recovery IDs of the transaction signature.
+     * @param r Ordered array of "r" output values of the ECDSA signatures.
+     * @param s Ordered array of "s" output values of the ECDSA signatures.
+     */
+    function delegateBySigs(
+        address delegatee,
+        uint256[] calldata nonce,
+        uint256[] calldata expiry,
+        uint8[] calldata v,
+        bytes32[] calldata r,
+        bytes32[] calldata s
+    )
+        external
+        override
+    {
+        require(
+            nonce.length == expiry.length
+            && nonce.length == v.length
+            && nonce.length == r.length
+            && nonce.length == s.length,
+            "WGT::delegateBySig: Mismatched array lengths"
+        );
+
+        for (uint256 i = 0; i < nonce.length; i++) {
+            delegateBySig(delegatee, nonce[i], expiry[i], v[i], r[i], s[i]);
+        }
+    }
+
+    /**
+     * @dev Delegates votes from one address to another.
+     * @param delegator Address delegating their votes.
+     * @param delegatee Address votes are delegated to.
+     */
+    function _delegate(address delegator, address delegatee) internal {
+        address currentDelegate = delegates[delegator];
+        uint256 delegatorBalance = balanceOf(delegator);
+        delegates[delegator] = delegatee;
+
+        emit DelegateChanged(delegator, currentDelegate, delegatee);
+
+        _moveDelegates(currentDelegate, delegatee, delegatorBalance);
+    }
+
+    /**
+     * @dev Moves votes from one account to another.
+     * @param srcRep Address of account to move votes from.
+     * @param dstRep Address of account to move votes to.
+     * @param amount Number of votes to move.
+     */
+    function _moveDelegates(
+        address srcRep,
+        address dstRep,
+        uint256 amount
+    )
+        internal
+    {
+        if (srcRep != dstRep && amount > 0) {
+            if (srcRep != address(0)) {
+                uint256 srcRepNum = numCheckpoints[srcRep];
+                uint256 srcRepOld = srcRepNum > 0 ? checkpoints[srcRep][srcRepNum - 1].votes : 0;
+                uint256 srcRepNew = srcRepOld - amount;
+
+                _writeCheckpoint(srcRep, srcRepNum, srcRepOld, srcRepNew);
+            }
+
+            if (dstRep != address(0)) {
+                uint256 dstRepNum = numCheckpoints[dstRep];
+                uint256 dstRepOld = dstRepNum > 0 ? checkpoints[dstRep][dstRepNum - 1].votes : 0;
+                uint256 dstRepNew = dstRepOld + amount;
+
+                _writeCheckpoint(dstRep, dstRepNum, dstRepOld, dstRepNew);
+            }
+        }
+    }
+
+    function _writeCheckpoint(
+        address delegatee,
+        uint256 nCheckpoints,
+        uint256 oldVotes,
+        uint256 newVotes
+    )
+        internal
+    {
+        uint256 blockNumber = block.number;
```

Comment:
is there a particular reason why Comp does a safe32 check here?

Reply:
They store the block number as a `uint32` value. Here it's just a `uint256`.